engine/server/engine/server/enclave_rest_api_handler.go

```diff
@@ -0,0 +1,857 @@
+
+// (GET /enclaves/{enclave_identifier}/artifacts)
+func (manager *enclaveRuntime) GetEnclavesEnclaveIdentifierArtifacts(ctx context.Context, request api.GetEnclavesEnclaveIdentifierArtifactsRequestObject) (api.GetEnclavesEnclaveIdentifierArtifactsResponseObject, error) {
+	enclave_identifier := request.EnclaveIdentifier
+	apiContainerClient, responseErr := manager.GetApiClientOrResponseError(enclave_identifier)
+	if responseErr != nil {
+		return api.GetEnclavesEnclaveIdentifierArtifactsdefaultJSONResponse{Body: *responseErr, StatusCode: int(responseErr.Code)}, nil
+	}
+
+	artifacts, err := (*apiContainerClient).ListFilesArtifactNamesAndUuids(ctx, &emptypb.Empty{})
+	if err != nil {
+		return nil, err
+	}
+
+	results := utils.MapList(
+		artifacts.FileNamesAndUuids,
+		func(x *rpc_api.FilesArtifactNameAndUuid) api_type.FileArtifactReference {
+			return api_type.FileArtifactReference{
+				Name: x.FileName,
+				Uuid: x.FileUuid,
+			}
+		})
+
+	return api.GetEnclavesEnclaveIdentifierArtifacts200JSONResponse(results), nil
+}
+
+// (POST /enclaves/{enclave_identifier}/artifacts/local-file)
+func (manager *enclaveRuntime) PostEnclavesEnclaveIdentifierArtifactsLocalFile(ctx context.Context, request api.PostEnclavesEnclaveIdentifierArtifactsLocalFileRequestObject) (api.PostEnclavesEnclaveIdentifierArtifactsLocalFileResponseObject, error) {
+	enclave_identifier := request.EnclaveIdentifier
+	apiContainerClient, responseErr := manager.GetApiClientOrResponseError(enclave_identifier)
+	if responseErr != nil {
+		return api.PostEnclavesEnclaveIdentifierArtifactsLocalFiledefaultJSONResponse{Body: *responseErr, StatusCode: int(responseErr.Code)}, nil
+	}
+	logrus.Infof("Uploading file artifact to enclave %s", enclave_identifier)
+
+	uploaded_artifacts := map[string]api_type.FileArtifactReference{}
+	for {
+		// Get next part (file) from the the multipart POST request
+		part, err := request.Body.NextPart()
+		if err == io.EOF {
+			break
+		}
+		filename := part.FileName()
+
+		client, err := (*apiContainerClient).UploadFilesArtifact(ctx)
+		if err != nil {
+			return nil, stacktrace.Propagate(err, "Can't start file upload gRPC call with enclave %s", enclave_identifier)
+		}
+		clientStream := grpc_file_streaming.NewClientStream[rpc_api.StreamedDataChunk, rpc_api.UploadFilesArtifactResponse](client)
+
+		response, err := clientStream.SendData(
+			filename,
+			part,
+			0, // Length unknown head of time
+			func(previousChunkHash string, contentChunk []byte) (*rpc_api.StreamedDataChunk, error) {
+				return &rpc_api.StreamedDataChunk{
+					Data:              contentChunk,
+					PreviousChunkHash: previousChunkHash,
+					Metadata: &rpc_api.DataChunkMetadata{
+						Name: filename,
+					},
+				}, nil
+			},
+		)
+
+		// The response is nil when a file artifact with the same has already been uploaded
+		// TODO (edgar) Is this the expected behavior? If so, we should be explicit about it.
```

Comment:
I think we should be explicit about it, probably a 400 status code

Reply:
+1, I added a per-file response using a 409